spec/models/api_request_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "rails_helper"
+require "fake_issues"
+
+RSpec.describe ApiRequest do
+  describe "#response" do
+    describe "with valid host, path, and query" do
+      it "returns an object with a code and a body" do
+        # stub the net/http request / response
+        success = Net::HTTPSuccess.new(1.0, "200", "OK")
+        allow_any_instance_of(ApiRequest)
+          .to receive(:send_api_request)
+          .and_return success
+        expect(success).to receive(:body).and_return('{"meta":{},"data":[]}')
+
+        # mock in host, path, and query
+        allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
+        allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
+        allow_any_instance_of(ApiRequest).to receive(:query).and_return "?some=query"
+
+        res = ApiRequest.new.response
+        expect(res).to match({code: "200", body: {"meta" => {}, "data" => []}})
+      end
+
+      it "is memoized" do
+        api = ApiRequest.new
+        expect(api).to receive(:get_response).once.and_return rand
```

Comment:
what's rand?

Reply:
This made more sense when it was `rand()` but our linter didn't like the parentheses.  I just made a random number to match up later, but also I can drop the comparison and just rely on the `.once` part instead, easily enough.

Reply:
maybe have it return a hast that is like:
{ random_test_response": rand } or something to make it clearer what's happening?

Reply:
I like @Sgtpluck's idea, and also, there's a chance that this will return the same random value no matter how many times it's called. You likely need to put that into a block for it to be re-evaluated each time `get_response` is called.

Reply:
good points!  I will do that!